### PR TITLE
Added Http code overwrite policy

### PR DESCRIPTION
**Pull request by palmieric**
Added a policy to overwrite the HTTP code returned from upstream.

### CODE REVIEW

**Comment by palmieric** [comment]
cfr. https://issues.redhat.com/browse/THREESCALE-6255

**Review by eloycoto — APPROVED** [review]

File: gateway/src/apicast/policy/statuscode_overwrite/code_overwrite.lua [hunk 1]
```diff
@@ -6,15 +6,14 @@ local new = _M.new
 function _M.new(config)
     local self = new(config)
     self.http_codes = {}
-    for _, code in ipairs(config.http_codes) do
+    for _, code in ipairs(config.http_codes or {}) do
```
Comment (palmieric) [inline gateway/src/apicast/policy/statuscode_overwrite/code_overwrite.lua:9]: @eloycoto once you are revising it, isn'it better to have consistent variables name?
`http_codes --> http_statuses`

File: spec/policy/statuscode_overwrite/statuscode_overwrite_spec.lua [hunk 2]
```diff
@@ -0,0 +1,51 @@
+local StatusCodeOverwrite = require('apicast.policy.statuscode_overwrite')
+
+describe('IP Check policy', function()
```
Comment (davidor) [inline spec/policy/statuscode_overwrite/statuscode_overwrite_spec.lua:3]: Wrong policy name

File: gateway/src/apicast/policy/statuscode_overwrite/code_overwrite.lua (hunk 1 above)
Comment (eloycoto) [reply gateway/src/apicast/policy/statuscode_overwrite/code_overwrite.lua:9]: Already updated on the jsonschema for the users ;-) 
https://github.com/3scale/APIcast/pull/1236/files#diff-d31ef981b73521817e16874af19636f5342d6791759acb1d77649d7eb2fff071R41

Comment (palmieric) [reply gateway/src/apicast/policy/statuscode_overwrite/code_overwrite.lua:9]: I mean:
```
self.http_codes = {}
    for _, code in ipairs(config.http_codes or {}) do
```

to 
```
self.http_statuses = {}
    for _, code in ipairs(config.http_statuses or {}) do
```

**Pull request merged**